lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/LibertyCompletionParticipant.java

```diff
@@ -53,12 +54,33 @@ public void onAttributeValue(String valuePrefix, ICompletionRequest request, ICo
         Properties variableProps = SettingsService.getInstance()
                 .getVariablesForServerXml(request.getXMLDocument()
                         .getDocumentURI());
-        String variableName = valuePrefix.replace("$", "")
-                .replace("{", "")
+        //getting all existing variables in current completion prefix string
+        List<VariableLoc> variables = LibertyUtils.getVariablesFromTextContent(valuePrefix);
```

Comment:
I went and looked at what this method is returning and noticed a couple of things that I should have commented previously. The two constants for the regex and pattern should have names that reflect what they are and should be all capital. See the `EXCLUDE_PATTERN_REGEX` and `EXCLUDE_PATTERN` constants in LibertyUtils. Change the names of `regex` and `p` accordingly. 

Now a quick question....is the getVariablesFromTextContent method returning just what was inside `${...}` without the beginning and ending syntax?

Reply:
Corrected variable names.
getVariablesFromTextContent returns variable value inside ${} and start and end index